### PR TITLE
Traceability and Frameproof and Triangle Codes

### PR DESCRIPTION
**Pull request by raleyr**

### CODE REVIEW

File: codes/quantum/qubits/surface/triangle.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+code_id: triangle
+physical: qubits
+logical: qubits
+
+name: 'Triangle code'
+introduced: '\cite{doi:10.22331/q-2017-04-25-2}'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:6]: arXiv:1612.04795

File: codes/quantum/qubits/surface/triangle.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+description: 'Triangle codes are types of non-CSS topological quantum error-correction codes with modified twist defects. A twist is a check of weight five that is embedded into the lattice of the code. For the triangle codes, the modified twist defects have weight four checks. Additionally, the codes have simple syndrome extraction circuits nad have a high threshold close to that of surface codes. Triangle codes also use about \(25\%\) fewer physical per logical qubit compared to the surface code \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:8]: use 25% fewer for a given distance? or something else?

File: codes/quantum/qubits/surface/triangle.yml (hunk 1 above)
Comment (valbert4) [reply codes/quantum/qubits/surface/triangle.yml:6]: Try to use arxiv links when available

File: codes/quantum/qubits/surface/triangle.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+protection: 'Protects against depolarizing noise. In particular, with \(13\) physical qubits triangle codes have a high-pseudothreshold fault-tolerance for depolarizing noise \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:10]: Is there a [[n,k,d]] notation for this code? If so, please list here.

File: codes/quantum/qubits/surface/triangle.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+features:
+
+  rate: 'For specific triangle codes, the rates are \(7/13\) or \(7/15\) both with distance \(3\) and weight-four check operators. In general, for \(d\) distance, there are \(3d^2 + O(d)\), \(9d^2/4 + O(d)\), or \(6d^2/4 + O(d)\) physical qubits per logical qubit depending on the type of initialization and measurement procedures (code conversion vs. basis-state conversion vs. CAT states). The general triangle codes still have weight-four check operators \cite{doi:10.22331/q-2017-04-25-2}.'
+
+  encoders:
+    - 'The general structure is to tile the plane with equilateral triangles. Each of these triangles can encode a qubit where one side is a string of \(Z\)s as \(\bar{Z}\), one side is a strong of \(X\)s as \(\bar{X}\), and the last side is a mixed-type string as \(\bar{Y}\). The size of the triangular patches and which patch encodes data versus act as ancillas for gates depends on the initialization and measurement procedures. Lastly, the triangles have a twist defect in the center \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:17]: Move first bullet into description.

File: codes/quantum/qubits/surface/triangle.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  transversal_gates: 'Triangle codes have transversal single-qubit Clifford gates of order \(3\) like Clifford gate \(\bar{SH}\) that performs \(H\) on all qubit positions \(x > 0\) or at \(x = 0\) and \(z > 0\) and then perform \(SH\) on the central qubit. Note that qubits are places on integer lattics in the \(xy-, yz-,\) and \(xz-\) planes \cite{doi:10.22331/q-2017-04-25-2}.'
+
+  #general_gates:
+
+  decoders:
+    - 'For decoding a single layer of the triangle code, a decoding graph is laid on the code. Nodes of the graph are located at each stabilizer (center of the triangle graph) and has red or blue edges where red associates with \(X\) errors and blue with \(Z\) errors. To take into account any errors from measuring the error syndrome, a three-dimensional stack of the decoding graphs is laid ontop of the code with vertical edges connecting to qubits between layers \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:27]: Any comments on decoding complexity would be great to add here. I'm assuming it's comparable to surface code?

File: codes/quantum/qubits/surface/triangle.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  fault_tolerance:
+    - 'The symmetry of triangle codes allows for fault-tolerant measurement and initialization in any Pauli basis \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:30]: "encoding in any logical Pauli basis" (at least I think this is our convention of what you mean)

File: codes/quantum/qubits/surface/triangle.yml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    - 'A non-fault-tolerant curcuit initializes the triangle code then to guarantee fault-tolerance, this is followed by postselection on trivial measurements of the syndrome and of the appropriate logical Pauli based on the basis of the logical states \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:31]: Run-on sentence

File: codes/quantum/qubits/surface/triangle.yml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    - 'For syndrome extraction, instead of using the conventional ordering of coupling to the loop after coupling to the plaquette, the extraction has the coupling to the loop interwoven with the coupling to the plaquette. These are equivalent in circuits, but makes a different to the fault-tolerance since the latter can detect hook errors and the former cannot \cite{doi:10.22331/q-2017-04-25-2}.'
+
+  code_capacity_threshold:
+    - '\(10\%\) bit-flip error-correction threshold for ideal syndrome measurements and \(10\%\) bit-phase flip error-correction threshold for ideal syndrom measurements \cite{doi:10.22331/q-2017-04-25-2}.'
+    - '\(3.2\%\) bit-flip error-correction threshold for noisy syndrome measurements and \(2.6\%\) for bit-phase flip noise \cite{doi:10.22331/q-2017-04-25-2}.'
+
+  threshold:
+    - 'For a triangle code with 7 qubits of information, 6 ancilla qubits, 13 qubits in total, and max degree of 4 the lower pseudothreshold is \(1.05 \times 10^{-4}\) and upper pseudothreshold is \(1.22 \times 10^{-4}\) \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:39]: Define pseudothreshold and combine into one bullet point a la "For a triangle code with 7 qubits of information, 6 (6, 8) ancilla qubits, 13 (13,15) qubits in total, and max degree of 4 (5, 4), the lower pseudothreshold is \(1.05 \times 10^{-4}\) (\\(...\\), \\(...\\))..."

File: codes/quantum/qubits/surface/triangle.yml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    - 'For a triangle code with 7 qubits of information, 6 ancilla qubits, 13 qubits in total, and max degree of 5 the lower pseudothreshold is \(1.57 \times 10^{-4}\) and upper pseudothreshold is \(1.92 \times 10^{-4}\) \cite{doi:10.22331/q-2017-04-25-2}.'
+    - 'For a triangle code with 7 qubits of information, 8 ancilla qubits, 15 qubits in total, and max degree of 4 the lower pseudothreshold is \(1.76 \times 10^{-4}\) and upper pseudothreshold is \(2.23 \times 10^{-4}\) \cite{doi:10.22331/q-2017-04-25-2}.'
+
+realizations:
+  - 'Code used to improve surface codes by creating a way to implement gate \(S\) without state distillation \cite{doi:10.22331/q-2017-04-25-2}.'
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:44]: This is a perfect sentence for early on in the description.

File: codes/quantum/qubits/surface/triangle.yml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+notes:
+  - 'Several tables and helpful figured to more easily see what the code looks like can be found in \cite{doi:10.22331/q-2017-04-25-2}.'
+
+relations:
+  parents:
+    - code_id: surface #not sure on this since these are CSS codes, but triangle are non-CSS
```
Comment (valbert4) [inline codes/quantum/qubits/surface/triangle.yml:51]: change parent to clifford-deformed_surface, and mention that these are non-CSS derivatives of the surface code.

**Comment by valbert4** [comment]
Great submission, thanks. Please address the minor comments and resubmit all codes, making sure things compile.

File: codes/quantum/qubits/surface/triangle.yml (hunk 3 above)
Comment (raleyr) [reply codes/quantum/qubits/surface/triangle.yml:10]: There was not one provided, sadly.

**Pull request closed without merging**